lib/charms/mongodb/v0/set_status.py

```diff
@@ -153,7 +153,7 @@ def is_unit_status_ready_for_upgrade(self) -> bool:
         if isinstance(current_status, ActiveStatus):
             return True
 
-        if not isinstance(current_status, WaitingStatus):
+        if not isinstance(current_status, BlockedStatus):
```

Comment:
This function is only called when it's a shard as you can see first by the `status_message` that is checked afterward. The case you are sending happens only when the charm is running as `config server`.
I think it might be valid here to check again that it's a `shard` to avoid future bugs if this function starts being used elsewhere on a config server for example.

Reply:
> I think it might be valid here to check again that it's a shard to avoid future bugs if this function starts being used elsewhere on a config server for example.

Yes this would be good to add. If you're able to add it now that would be great. Good eyes. If not we can revisit